sockeye/model.py

```diff
@@ -317,6 +317,17 @@ def load_parameters(self,
                                 cast_dtype=cast_dtype, dtype_source=dtype_source)
         logger.info('Loaded params from "%s" to "%s"', filename, mx.cpu() if ctx is None else ctx)
 
+    def set_parameters(self, new_params: dict):
+        """
+        Update model params with new values from a dictionary.
+
+        :param new_params: Dictionary containing the new parameters.
+        """
+        model_params = self.collect_params()
+        for k in new_params:
+            if k in model_params:
+                model_params[k].set_data(new_params[k].data())
```

Comment:
I assume this would fail if `model_params[k]` or `new_params[k]` are not initialized yet. Can you add a unit test for this? Would it make sense to fail here instead of relying on the implementation of `set_data`?

Reply:
Thanks for the suggestion! I added failures here and some unit tests to address this.